lib/puppet/pops/lookup/lookup_adapter.rb

```diff
@@ -68,7 +68,8 @@ def lookup(key, lookup_invocation, merge)
   end
 
   def lookup_global(key, lookup_invocation, merge_strategy)
-    terminus = Puppet[:data_binding_terminus]
+    # hiera_xxx will always use global_provider regardless of data_binding_terminus setting
+    terminus = lookup_invocation.hiera_xxx_call? ? :hiera : Puppet[:data_binding_terminus]
```

Comment:
This PR fixes the regression that `hiera_xxx()` functions was affected by the `data_binding_terminus`. This means that the behavior after it's applied is as it where in 4.8.x. That's what's needed in order to fully address the problem described in the ticket.

If we modify lookup so that it behaves differently depending on if the change stems from a `lookup()` call or an APL, then we introduce a breaking change.